grid_map_costmap_2d/CMakeLists.txt

```diff
@@ -32,13 +23,44 @@ set(dependencies
   tf2_geometry_msgs
 )
 
+add_library(${PROJECT_NAME}
+  include/grid_map_costmap_2d/costmap_2d_converter.hpp
+  include/grid_map_costmap_2d/grid_map_costmap_2d.hpp
+)
+
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
```

Comment:
Namespace is added in `ament_export_target`. Is this alias necessary?

Reply:
This is used in the BUILD_INTERFACE when I link this library to tests. Without an ALIAS to link to, linking tests to the executable is ripe for typos in the name, and you would instead get include or link errors. With linking to an ALIAS target internally, it then changes to a more informative error during configuration time.

 This is mentioned in the CMP0028 docs:
https://cmake.org/cmake/help/latest/policy/CMP0028.html#policy:CMP0028

Reply:
See the usage here; this is where you link to the ALIAS target. 
```cmake
target_link_libraries(costmap-2d-ros-test
  ${PROJECT_NAME}::${PROJECT_NAME}
)
```